src/instana/instrumentation/sanic_inst.py

```diff
@@ -1,144 +1,125 @@
-        async def stream_with_instana(wrapped, instance, args, kwargs):
-            response = args[0]
-            span = async_tracer.active_span
-            if span is None:
-                await wrapped(*args, **kwargs)
-            else:
-                response_details(span=span, response=response)
-                try:
-                    await wrapped(*args, **kwargs)
-                except Exception as exc:
-                    span.log_exception(exc)
-                    raise
-
-
-    @wrapt.patch_function_wrapper('sanic.app', 'Sanic.handle_request')
-    async def handle_request_with_instana(wrapped, instance, args, kwargs):
-
-        try:
-            request = args[0]
-            try:  # scheme attribute is calculated in the sanic handle_request method for v19, not yet present
+    from typing import Callable, Tuple, Dict, Any
+    from sanic.exceptions import SanicException
+
+    from opentelemetry import context, trace
+    from opentelemetry.trace import SpanKind
+    from opentelemetry.semconv.trace import SpanAttributes
+
+    from instana.log import logger
+    from instana.singletons import tracer, agent
+    from instana.util.secrets import strip_secrets_from_query
+    from instana.util.traceutils import extract_custom_headers
+    from instana.propagators.format import Format
+
+    from sanic.request import Request
+    from sanic.response import HTTPResponse
+
+    @wrapt.patch_function_wrapper("sanic.app", "Sanic.__init__")
+    def init_with_instana(
+        wrapped: Callable[..., sanic.app.Sanic.__init__],
+        instance: sanic.app.Sanic,
+        args: Tuple[object, ...],
+        kwargs: Dict[str, Any],
+    ) -> None:
+        wrapped(*args, **kwargs)
+        app = instance
+
+        @app.middleware("request")
+        def request_with_instana(request: Request) -> None:
+            try:
                 if "http" not in request.scheme:
-                    return await wrapped(*args, **kwargs)
-            except AttributeError:
-                pass
-            headers = request.headers.copy()
-            ctx = async_tracer.extract(opentracing.Format.HTTP_HEADERS, headers)
-            with async_tracer.start_active_span("asgi", child_of=ctx) as scope:
-                scope.span.set_tag('span.kind', 'entry')
-                scope.span.set_tag('http.path', request.path)
-                scope.span.set_tag('http.method', request.method)
-                scope.span.set_tag('http.host', request.host)
+                    return
+
+                headers = request.headers.copy()
+                parent_context = tracer.extract(Format.HTTP_HEADERS, headers)
+
+                span = tracer.start_span("asgi", span_context=parent_context)
+                request.ctx.span = span
+
+                ctx = trace.set_span_in_context(span)
+                token = context.attach(ctx)
+                request.ctx.token = token
+
+                span.set_attribute("span.kind", SpanKind.CLIENT)
```

Comment:
Should it not be `SpanKind.SERVER` since this is a web framework and responds to entry spans?
```suggestion
                span.set_attribute("span.kind", SpanKind.SERVER)
```

Reply:
You're right, its an entry span. Changed it now.